lib/matchers/indexes.rb

```diff
@@ -0,0 +1,84 @@
+module Mongoid
+  module Matchers
+    class HaveIndexForMongoid3 # :nodoc:
```

Comment:
This class should be the same name for different versions. Make `lib/matchers/indexes/v3/have_index_for.rb` and `lib/matchers/indexes/v4/have_index_for.rb` for example and include the right file. This will reduce the switching.